docs/design/proposals/rbd-mirror.md

```diff
@@ -0,0 +1,73 @@
+# RBD MIRRORING
+
+RBD mirroring is a process of replication of RBD images between two or more
+Ceph clusters. Mirroring ensures point-in-time, crash-consistent RBD images
+between clusters, RBD mirroring is mainly used for disaster recovery (i.e.
+having a secondary site as a failover). See Upstream [Ceph
+documentation](https://docs.ceph.com/en/latest/rbd/rbd-mirroring) on RBD
+mirroring for complete information.
+
+## Architecture
+
+![mirror](rbd-mirror.png)
+
+## Design
+
+Currently, ceph-csi generates its unique ID for each RBD image and stores the
+corresponding PV name and the unique ID mapping, It creates the RBD image with
+the unique ID and returns the encoded value which contains all the required
+information for other operations. For mirroring, the same RBD image will be
+mirrored to the secondary cluster. As the journal(omap data) is not mirrored to
+the secondary cluster, The cephcsi cannot identify the RBD image corresponding
+to the PV wihtout omap data.
+
+**Note:** It's expected that the user/admin will create the static PVC's on the
+secondary site during the failover.
+
+**Note:** when the static PVC created on the secondary site we cannot use the
+volumeHandle to identify the omap data or the image anymore because as we have
+only PoolID and clusterID in the volumeHandle. we cannot identify the correct
+pool name from the poolID because pool name will remain the same on both clusters
+but not the poolID even the clusterID can be different on the secondary
+cluster.
+
+> **VolumeHandle** is the unique volume name returned by the CSI volume plugin’s
+CreateVolume to refer to the volume on all subsequent calls.
+
+Once the static PVC is created on the secondary cluster, the user can try
+delete the PVC,expand the PVC or mount the PVC. Incase of mounting (node stage)
+we will get the volume context in GRPC call but not in the Delete/Expand
```

Comment:
Do not call them GRPC calls, that is a CSI internal implementation detail. Try to stick to the naming of operations or *CSI RPCs* as the spec does. (GRPC is just the format, *CSI RPC*, or *RPC call* is better in this doc)